desktop/cmp/treemap/TreeMap.js

```diff
@@ -0,0 +1,297 @@
+/*
+ * This file belongs to Hoist, an application development toolkit
+ * developed by Extremely Heavy Industries (www.xh.io | info@xh.io)
+ *
+ * Copyright © 2019 Extremely Heavy Industries Inc.
+ */
+import {Component} from 'react';
+import PT from 'prop-types';
+import {Highcharts, highchartsExporting, highchartsOfflineExporting, highchartsExportData, highchartsTree, highchartsHeatmap} from '@xh/hoist/kit/highcharts';
+
+import {XH, elemFactory, HoistComponent, LayoutSupport} from '@xh/hoist/core';
+import {div, box} from '@xh/hoist/cmp/layout';
+import {Ref} from '@xh/hoist/utils/react';
+import {resizeSensor} from '@xh/hoist/kit/blueprint';
+import {fmtNumber} from '@xh/hoist/format';
+import {forEachAsync} from '@xh/hoist/utils/async';
+import {Cube} from '@xh/hoist/data/cube';
+import {assign, merge, clone, debounce} from 'lodash';
+
+import {LightTheme} from './theme/Light';
+import {DarkTheme} from './theme/Dark';
+
+import './TreeMap.scss';
+import {TreeMapModel} from './TreeMapModel';
+
+highchartsExporting(Highcharts);
+highchartsOfflineExporting(Highcharts);
+highchartsExportData(Highcharts);
+highchartsTree(Highcharts);
+highchartsHeatmap(Highcharts);
+
+/**
+ * Component for rendering a TreeMap.
+ *
+ * It is a managed wrapper around a Highcharts TreeMap visualization, which renders
+ * records from a Store and optionally binds to a GridModel.
+ *
+ * @see TreeMapModel
+ */
+@HoistComponent
+@LayoutSupport
+export class TreeMap extends Component {
+
+    static propTypes = {
+        /** Primary component model instance. */
+        model: PT.oneOfType([PT.instanceOf(TreeMapModel), PT.object]).isRequired
+    };
+
+    static modelClass = TreeMapModel;
+
+    baseClassName = 'xh-treemap';
+
+    _chartElem = new Ref();
+    _chart = null;
+
+    constructor(props) {
+        super(props);
+
+        // Detect double-clicks vs single-clicks
+        this._clickCount = 0;
+        this._debouncedClickHandler = debounce(this.clickHandler, 500);
+
+        // Sync selection
+        this.addReaction({
+            track: () => [this.model.selectedIds, this.getMergedConfig()],
+            run: () => this.syncSelection(),
+            delay: 1 // Must wait for chart re-render on data / config change
+        });
+    }
+
+    render() {
+        // Default flex = 1 (flex: 1 1 0) if no dimensions / flex specified, i.e. do not consult child for dimensions;
+        const layoutProps = this.getLayoutProps();
+
+        if (layoutProps.width == null && layoutProps.height == null && layoutProps.flex == null) {
+            layoutProps.flex = 1;
+        }
+
+        this.renderHighChart();
```

Comment:
Is this a no-op on first render as the div has yet to be setup on the DOM?  Just thinking that through - is it the ref getting its value that then triggers it to loop back through?  Might be worth a comment as it seems kind of unusual (although understandable).